### PR TITLE
refactor: modernize codebase, remove Angular and legacy dependencies

### LINKED ISSUE #42: `$section` in main.js is evaluated before its corresponding element exists in the DOM
> `$section` in main.js is evaluated before its corresponding element exists in the DOM

As a result, the functionality associated with the would-be element never executes due a `return` statement that's guarded by a conditional which always evaluates to `true`:

https://github.com/m-lab/mlab-speedtest/blob/4cb40a00cc59ad46f8c7c1912461614c2bdc40a7/app/assets/js/main.js#L93-L94

### PR DESCRIPTION
**Pull request by robertodauria**
## Summary

- Remove Angular 1.x, jQuery, Gulp, Bower, Skel framework
- Replace with vanilla JS (app.js, gauge.js, i18n.js)
- Add runtime i18n with `?lang=` query param support
- Simplify CSS from ~4000 to ~900 lines
- Add simple Node.js build script (scripts/build.js)
- Convert .po translations to JSON format
- Update GitHub Actions workflows for new build
- Change Firebase public dir from app/ to dist/

Dependencies reduced from ~680 to 13 packages.

Closes #42, #40, #36, #111, #147, #152, #159

### CODE REVIEW

File: src/js/app.js [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    let token = null;
+    try {
+      const tokenResp = await fetch(tokenURL);
+      const tokenData = await tokenResp.json();
+      token = tokenData.token;
+    } catch (err) {
+      console.warn('Failed to fetch token, running without priority access:', err);
+    }
+
+    const self = this;
```
Comment (bassosimone) [inline src/js/app.js]: Why do we need this? (I find this assignment a bit tricky w/o explanation)

**Comment by Kunal1522** [comment]
Hi @robertodauria and @bassosimone,

I was wondering if it would be okay to start planning the unit test cases based on the current state of this PR, even though it might be merged over the weekend.

I feel that outlining the test cases early could be helpful in building a stronger foundation for the GSoC project. although, I’m happy to wait or adjust based on your guidance.

File: src/js/app.js [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+              data.Data.ElapsedTime / (this.TIME_EXPECTED * 2);
+            ProgressGauge.progress(progress);
+          }
+        },
+        downloadComplete: (data) => {
+          this.measurementResult.s2cRate =
+            data.LastClientMeasurement.MeanClientMbps.toFixed(2) + ' Mb/s';
+          this.measurementResult.latency =
+            (data.LastServerMeasurement.TCPInfo.MinRTT / 1000).toFixed(0) + ' ms';
+          this.measurementResult.loss =
```
Comment (anuzx) [inline src/js/app.js]: there may be a division by zero issue if ByteSent is 0 (which can be due to metadata error or early test fail) while calculating packet loss , this may return in NaN% being displayed , should we guard against this edge case to ensure stable UI behavior?
@robertodauria

File: src/js/app.js [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+            (data.LastServerMeasurement.TCPInfo.BytesRetrans /
+              data.LastServerMeasurement.TCPInfo.BytesSent * 100).toFixed(2) + '%';
```
[the remark this answers is not in the document]
Comment (robertodauria) [reply src/js/app.js]: Hi @anuzx, this PR is not open for external review.

This exact same suggestion was already submitted as PR #156 and closed — `BytesSent` cannot be zero in this code path. By the time the measurement callback executes, the client has already completed TCP, TLS, and WebSocket handshakes. If `BytesSent` were zero, we would never have received the server measurement message in the first place.

If you are interested in contributing to M-Lab — including for GSoC — please read our [contribution requirements](https://github.com/m-lab/gsoc#2-engage-with-us) first. We expect contributors to coordinate with maintainers before starting work, and to demonstrate genuine understanding of the codebase. Leaving unsolicited review comments on PRs with assigned reviewers is not a productive way to engage with the project.

File: package.json [hunk 4]
```diff
@@ -1,47 +1,22 @@
...
-    "jasmine-core": "^2.4.1",
-    "karma": "^0.13.22",
-    "karma-chrome-launcher": "^0.2.3",
-    "karma-firefox-launcher": "^0.1.7",
-    "karma-jasmine": "^0.3.8",
-    "karma-junit-reporter": "^0.4.1",
-    "protractor": "^3.2.2",
-    "shelljs": "^0.6.0"
-  },
+  "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (robertodauria) [reply package.json]: The msak client is now initialized as speed-measurementlab-net `v1.0.0` consistently with the package version.
